core/management/commands/import_facility_data.py

```diff
@@ -78,16 +77,127 @@ def load_ccn_data_to_facility_model(self, export_path, care_type):
+        else:
+            caphs_df = self.extract_questions_as_rows(caphs_df, care_type)
+        
+        caphs_df = caphs_df.drop_duplicates()
+        return caphs_df
+        
+    def create_caphs_json_by_row_of_all_caphs_df(self, all_cahps_df):
+        for index, row in all_cahps_df.iterrows():
+            # Convert the row to a dictionary
+            hospital = row.to_dict()  
+            # changing dict to json we need json type to save in the instance
+            hospital = json.dumps(hospital) 
+            if Facility.objects.filter(facility_id=row["Facility ID"]):
+                facility = Facility.objects.filter(facility_id=row["Facility ID"]).first()
+                caphs_metrics = CAPHSMetrics(
+                                    caphs_metric_json=hospital,
+                                    facility=facility
+                                    )
+                caphs_metrics.save()
+            else:
+                pass
+        
     def handle(self, *args, **options):
         export_path = DATA_DIR
-        # only using these two care types for now because other care types metrics are not properly defined
-        caphs_care_types = ["Home Health", "Outpatient Ambulatory Services"]
+        # load all ccn data into df and create facility for each row
         ccn_care_types = ["ED", "Home Health", "Hospice", "Hospital", "Outpatient"]
-        # care_types = ["Outpatient Ambulatory Services", "Home Health", "Hospice", "Hospitals", "Nursing Homes"]
-        # self.create_instance_for_each_hcaphs(export_path, care_types)
         for care_type in ccn_care_types:
-            print('care_type', care_type)
+            print('ccn care_type', care_type)
             self.load_ccn_data_to_facility_model(export_path, care_type)
         
-    
\ No newline at end of file
+        caphs_care_types = ["ED + Others", "Home Health", "Hospice", "Hospitals", "In-Center Hemodialysis", "Nursing Homes"]
+        files_with_measures_as_columns = ["Home Health", "Outpatient", "Nursing Homes", "In-Center Hemodialysis"]
+        all_cahps_df = pd.DataFrame()
+        #load all caphs data and merge them into one df
+        for care_type in caphs_care_types:
+            print('caphs care_type', care_type)
+            cur_cahps_df = self.load_caphs_data(export_path, care_type)
+            
+            #combine caph df into all_caphs_df
+            if any(file_substring in care_type for file_substring in files_with_measures_as_columns):
+                all_cahps_df = pd.concat([all_cahps_df, cur_cahps_df])
+            else:
+                if all_cahps_df.empty:
+                    all_cahps_df = cur_cahps_df
+                else:
+                    all_cahps_df = pd.merge(all_cahps_df, cur_cahps_df, how="outer", on="Facility ID")
+                    
+        all_cahps_df = all_cahps_df.reset_index(drop=True)
+
+        self.create_caphs_json_by_row_of_all_caphs_df(all_cahps_df)
```

Comment:
hard to understand this function name

Reply:
how does `populate_caph_json_by_df_row`?